src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -86,38 +69,69 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
 }, {});
 
 const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelect }) => {
-  const [isNewSelectedPath, setIsNewSelectedPath] = useState(false);
-  const [previousSelectedPath, setPreviousSelectedPath] = useState(selectedPath);
   const [openKeys, setOpenKeys] = useState(openKeysToItem(menuItems[0], selectedPath));
+  const currentNodeId = useRef();
+  const cursor = useRef(0);
   const selectedItem = useRef();
+  const previousSelectedPath = useRef(selectedPath);
   const theme = useContext(ThemeContext);
+  const visibleNodes = [];
 
-  if (previousSelectedPath !== selectedPath) {
-    setIsNewSelectedPath(true);
-    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
-    setPreviousSelectedPath(selectedPath);
-  }
+  /**
+   * Assigns focus to current node.
+   */
+  const focusCurrentNode = () => {
+    const currentNode = currentNodeId.current ? document.getElementById(currentNodeId.current) : null;
+    if (currentNode) {
+      currentNode.focus();
+    }
+  };
 
+  /**
+   * Scrolls the currently selected menu item into view on mount.
+   * Ensures that the cursor is synched with the currently selected item.
+   */
   useEffect(() => {
+    const nodeId = selectedItem.current?.id;
+    const idx = visibleNodes.findIndex((el) => el.id === nodeId);
+    if (idx >= 0) {
+      cursor.current = idx;
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
```

Comment:
We will need to ensure this doesn't occur on initial page load. We want the scrolling, but not the focus in that case. Focus should remain on document body until the users begins navigating the page.

Reply:
This took a while to get working, but it should be done now. A menu item will only receive focus when the navigation menu itself receives focus.